signers/legacy/search.go

```diff
@@ -0,0 +1,238 @@
+	}
+
+	// Request method determines what we hash
+    if r.Method == "POST" {
+		body, err := signers.ReadBody(r)
+		if err != nil {
+			return signers.Errorf(500, signers.ErrorTypeInternalError, "Failed to read request body: %s", err.Error())
+		} 
+	    hash = generateSignature(string(body), request_timestamp, secret)
+	    logger.Print("body: " + string(body))
+
+    } else {
+        path_and_query = r.URL.Path + "?" + r.URL.RawQuery
+        hash = generateSignature(path_and_query, request_timestamp, secret)
+        logger.Print("Path and Query: " + path_and_query)
+    }
+
+    if hash != auth_headers["acquia_solr_hmac"] {
+		logger.Print("Expected: ", hash)
+		logger.Print("Received: ", auth_headers["acquia_solr_hmac"])
+    	return signers.Errorf(403, signers.ErrorTypeInvalidRequiredHeader, "Hash in acquia_solr_hmac does not match expected value")
+    }
+    // All checks passed, request is authorized
+    return nil
+}
+
+func (v *SearchSigner) SignDirect(r *http.Request, authHeaders map[string]string, secret string) *signers.AuthenticationError {
+
+	var hash string
+	var path_and_query string
+	var request_time int64
+	var nonce string
+
+	request_time = time.Now().Unix()
+	nonce = getNonce()
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return signers.Errorf(500, signers.ErrorTypeInternalError, "Failed to read request body: %s", err.Error())
+	}
+
+    if r.Method == "POST" {
+	    hash = generateSignature(string(body), request_time, secret)
+	    logger.Print("body: " + string(body))
+    } else {
+        path_and_query = r.URL.Path + "?" + r.URL.RawQuery;
+        hash = generateSignature(path_and_query, request_time, secret);
+        logger.Print("Path and Query: " + path_and_query)
+    }
+
+    r.AddCookie(&http.Cookie{Name: "acquia_solr_time", Value: strconv.FormatInt(request_time, 10)})
+    logger.Print("request_time: " + strconv.FormatInt(request_time, 10))
+    r.AddCookie(&http.Cookie{Name: "acquia_solr_nonce", Value: nonce})
+    logger.Print("nonce: " + nonce) 
+    r.AddCookie(&http.Cookie{Name: "acquia_solr_hmac", Value: hash})
+    logger.Print("hash: " + hash)
+
+    return nil
+
+}
+
+func (v *SearchSigner) GenerateAuthorization(r *http.Request, authHeaders map[string]string, signature string) (string, *signers.AuthenticationError) {
+	//TODO: this function was added because signers.Signer requires it
+	return fmt.Sprintf("Search GenerateAuthorization"), nil
+}
+
+func (v *SearchSigner) GetIdentificationRegex() *regexp.Regexp {
+	return v.IdRegex
+}
+
+func getNonce() (string) {
+	//"github.com/dchest/uniuri"
+	//var char_list = []byte(" !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}")
+	//TODO: base64 encode before returning
+	//return NewLenChars(24, char_list)
+	return "ABCDEFGHIJKLMNOPQRSTUVWX"
```

Comment:
Ideally, legacy implementations match the current one on the service and then the service strives to migrate to one of the mainstream (v2) implementations. That's why this is merged as-is. While I don't know your inner workings, it would make sense not to worry further about the legacy implementation as long as it works.